File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int buy=prices[0];int sell=0;
        for(int i=0;i<prices.size();i++){
            if(prices[i]<buy) buy=prices[i];
            sell=max(sell,(prices[i]-buy));
        }
        return sell;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let buy=prices[0]; let mp=0;
    for(let i=1;i<prices.length;i++){
        if(buy>prices[i]){
            buy=prices[i];
        }
        else {
            mp=Math.max(mp,prices[i]-buy);
        }
    } 
    return mp;
};